src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import duke.exception.InvalidDeadlineException;
+import duke.exception.InvalidTaskException;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A class that deals with loading and saving data to/from a file.
+ *
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructor of a storage.
+     *
+     * @param filePath A path that access the location of the file to be open and used.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+
+    /**
+     * Method to load any existing data from file to the current list
+     *
+     * @return ArrayList of current tasks.
+     * @throws IOException thrown when file cannot be found.
+     * @throws InvalidTaskException thrown when users give invalid input.
+     * @throws InvalidDeadlineException thrown when users give invalid deadlines.
+     */
+
+    public ArrayList<Task> load() throws IOException,
+            InvalidTaskException, InvalidDeadlineException {
+        ArrayList<Task> list = new ArrayList<>();
+        File file = new File(filePath);
+
+        if (file.exists()) {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNextLine()) {
+                String str = sc.nextLine();
+                String[] stuff = str.split(" \\| ");
+                Task task;
+                switch (stuff[0]) {
+                case "T":
+                        task = new ToDos(stuff[2]);
+                        break;
+                case "D":
+                        task = new Deadline(stuff[2], stuff[3]);
+                        break;
+                case "E":
+                        task = new Events(stuff[2], stuff[3]);
+                        break;
+                default:
+                        throw new InvalidTaskException();
```

Comment:
Should there be a break statement here after throwing the InvalidTaskException?

```suggestion
                        throw new InvalidTaskException();
                        break
```

Reply:
I think there shouldn't be a break here since an exception is thrown? So the code wont be able to reach line 61 if im not wrong